controllers/userController.js

```diff
@@ -7,23 +11,52 @@ const getUsers = (req, res) => {
     res.json(users)
 };
 
-const createUser = (req, res) => {
-    const name = req.body.name;
-    const email = req.body.email;
-    const user = {
+const createUser = async (req, res) => {
+    try {
+        const salt = await bcrypt.genSalt();
+        const hashPassword = await bcrypt.hash(req.body.password, salt);
+        const name = req.body.name;
+        const email = req.body.email;
+        const password = hashPassword;
+        const user = {
         id: uuidv4(),
         email,
         name,
+        password
     };
-    if (!user.name) {return res.status(400).send('A name should be provided')}
+    if (!user.name) {res.status(400).send('A name should be provided')}
     users.push(user);
-    res.status(201).json(user);
+    res.status(201);
```

Comment:
@courage173 It has been a while I wrote express code. Is it possible to to `res.status(201);` here and still do `res.json({token})` on line 33? Last I remember, things like this cause the error `Cannot set headers after response has been sent` or something like that.

Reply:
didn't get such error tho, works just fine

Reply:
You were actually right concerning this error... I carefully did a quick run on postman, tried sending a post request to create a user without name and there is this code inside the if  statement that will be executed because there has to be a name yeah, but after providing a name and tried sending the post request again, I got the cannot set header error on the terminal

All I did was to return inside the if statement to stop that execution and I tried the above test again and it didn't throw such error again...